### ISSUE TITLE
Has the Image Classification model been successfully replicated?

### ISSUE THREAD

**Issue by plo97** [WHU-Sigma/HyperSIGMA#3]

**Comment by DotWang** [comment]
@plo97 do you mean whether the pretrained weight is loaded?

**Comment by DotWang** [comment]
@Maoabear Which dataset do you use? Do you finetuning or direct inference?

**Comment by Maoabear** [comment]
Hello,

I am using the notebook provided at [this link](https://github.com/WHU-Sigma/HyperSIGMA/blob/main/ImageClassification/demo%20cls%20hypersigma.ipynb). I have replaced the .pth file as mentioned earlier. Please refer to the attached document for further details.

Thank you.

**Comment by Maoabear** [comment]
Hello,

I am using the notebook you provided at [this link](https://github.com/WHU-Sigma/HyperSIGMA/blob/main/ImageClassification/demo%20cls%20hypersigma.ipynb). I have replaced the .pth file as mentioned earlier. 

The dataset should be the same as the default set in the notebook, the Indian one.

Thanks for your attention on this.

**Comment by DotWang** [comment]
@Maoabear How do you use the model weight, direct inference or training (finetuning)?

**Comment by Maoabear** [comment]
Hi, it seems neither direct-inference nor retraining work for me. 

I was just following the code from that jupyter notebook, which seems contain a traning cell indeed. I am looking forward to re-train the  weight from scratch and achieve the accuracy in reported in the paper.

Please see my code at 
https://github.com/Linsonng/HyperSIGMA/blob/main/ImageClassification/demo_cls_hypersigma.ipynb and your suggestion is appreciated.

**Comment by Maoabear** [comment]
Could you please provide the corresponding Xiongan Classification .ipynb for me to run directly and see the accuracy?

**Comment by Linsonng** [comment]
Hi @DotWang, would you mind pointing out how to reproduce the 85.54 classification accuracy on Indian Pines (shown on main/readme.md)? Would it be possible to train your classification model from scratch on Indian Pines to reach such results?

Thanks a lot for your help.

**Comment by jinyaoWHU** [comment]
@Maoabear We have uploaded the relevant files now, you can download and use them directly.

**Comment by jinyaoWHU** [comment]
@Linsonng Are you using demo seg hypersigma or demo cls hypersigma? 85.54 was obtained using demo seg hypersigma.

**Comment by Linsonng** [comment]
> @Linsonng Are you using demo seg hypersigma or demo cls hypersigma? 85.54 was obtained using demo seg hypersigma.

I am focusing on classification tasks. I am interested in the performance of your model on Indian Pines classification, can you tell me how to run on this? Thanks a lot!

**Comment by jinyaoWHU** [comment]
@Maoabear We have re uploaded the file to ensure that it matches the required files for demo cls hypersigma, and there is no need to import new files separately. And you can download the Xiong'an dataset from the following website http://www.hrs-cas.com/a/share/shujuchanpin/2019/0501/1049.html

**Comment by jinyaoWHU** [comment]
@Linsonng In our local experiments, classification accuracy is seriously affected by the type of normlization, and has low running efficiency. So we finally implement with segmentaion in the paper. The classification network is only used for the Xiongan dataset, see Table 15 in Section 5.7. You can directly run the demo seg hypersigma to obtain the results

**Comment by DotWang** [comment]
> Hi @DotWang, would you mind pointing out how to reproduce the 85.54 classification accuracy on Indian Pines (shown on main/readme.md)? Would it be possible to train your classification model from scratch on Indian Pines to reach such results?
> 
> Thanks a lot for your help.

please see the `readme` of the ImageClassification folder, we use the segmentation network.

**Comment by Linsonng** [comment]
<img width="249" alt="Screenshot 2024-07-26 at 15 40 45" src="https://github.com/user-attachments/assets/a85c5c0f-1e5d-4040-96f3-4a5ec89db282">

@jinyaoWHU Thanks for your patience, but I am just wondering is it possible and how to make the performance here in this figure reproducible.

**Comment by Linsonng** [comment]
@DotWang Yes thanks for the kind suggestion. I can only see one line in the ImageClassification readme file, am I referring to the right file? If that's right, is there any classification model available? Thanks for your great work

**Comment by DotWang** [comment]
@Linsonng of course, we have said we use the segmentaion network to achieve the accuracy of 85.54 on indian pines, What exactly do you want to ask？？？？？？？

It should be noted that in the hyperspectral community, the training samples are randomly selected, so the accuracy of the results may fluctuate, but the difference is not significant

**Comment by DotWang** [comment]
@Linsonng Up to now, we have not uploaded any finetuned model, so you may finetune it on the Indian Pines by initializing with the pretraining weights using `demo seg hypersigma.ipynb.`

**Comment by Linsonng** [comment]
> @Linsonng of course, we have said we use the segmentaion network to achieve the accuracy of 85.54 on indian pines, What exactly do you want to ask？？？？？？？
> 
> It should be noted that in the hyperspectral community, the training samples are randomly selected, so the accuracy of the results may fluctuate, but the difference is not significant

My interest is only in running classification task on Indian pines, and I come to this repo because I noticed that there is a figure in your paper saying so. A segmentation model is not my need. Thanks for your suggestion.

**Comment by DotWang** [comment]
@Linsonng Segmentaion can be seen image-level hyperspecral classification, it has the same output with the patch/pixel-level classification:

<img width="426" alt="1722007977450" src="https://github.com/user-attachments/assets/595346ae-3e3e-4d81-aa43-cc0fe8d9be88">

**Comment by zhangxiaopang88** [comment]
你好，想咨询一下，为什么对雄安数据做了分类？在雄安数据集上分类精度能达到多少？能否提供雄安数据上的分类模型？多谢

**Comment by DotWang** [comment]
@zhangxiaopang88 

雄安数据集规模很大，类别较多，和其它数据集比相对较难，所以我们主要用这个数据集检验模型的扩展性，来挖掘十亿模型真正的潜力，但是在这个数据集上训分割计算量比较大，就训分类了，这样也没有解码器之类的，更能体现模型的表征能力

**Comment by zhangxiaopang88** [comment]
> @zhangxiaopang88
> 
> 雄安数据集规模很大，类别较多，和其它数据集比相对较难，所以我们主要用这个数据集检验模型的扩展性，来挖掘十亿模型真正的潜力，但是在这个数据集上训分割计算量比较大，就训分类了，这样也没有解码器之类的，更能体现模型的表征能力
> 
> 分类精度论文给出来了，OA在90%以上

我复现你说的雄安数据集上的分类精度，分割的话跑起来了，但是很慢，按照你的经验是不是分割的结果会比分类的精度高一些

**Comment by DotWang** [comment]
@zhangxiaopang88 雄安数据集我们没跑分割的，很慢，直接拿分类跑的，我个人感觉分割的话效果会好点，但是这还会取决于你输入的图像块大小，还有vit的patch size，总之控制起来成本还是比较高的
